File: 단계별로 풀어보기/정수론 및 조합론/9357/패션왕 신해빈.py
import sys

def init():
    T = int(sys.stdin.readline())
    for i in range(T):
        t = int(sys.stdin.readline())
        closet = dict()
        result = 1
        for j in range(t):
            wearName, wearType = sys.stdin.readline().rstrip().split()
            if wearType in closet:
                closet[wearType] = closet[wearType]+1
            else:
                closet[wearType] = 1

        for k in closet.values():
            result = result*(k+1)

        sys.stdout.write('%d\n' %(result-1))
init()

File: 단계별로 풀어보기/정수론 및 조합론/9357/패션왕 신해빈.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int T, t, cnt;
    int result;
    string wearName, wearType;
    cin >> T;
    map<string,int> closet;
    
    for(int k=0; k < T; k++){
        closet.clear();
        cin >> t;
        result = 1;
        for(int i=0; i<t; i++){
            cin >> wearName >> wearType;
            if(closet.find(wearType)!=closet.end()){
                closet[wearType]++;
            }else{
                closet.insert({wearType,1});
            }
        }
        
        for(auto it = closet.begin(); it != closet.end(); it++){
            result *= it->second+1;    
        }
        cout << result-1<<'\n';
    }

    return 0;
}
